utils/zerovec/src/ule/macros.rs

```diff
@@ -0,0 +1,28 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[macro_export]
+macro_rules! impl_ule_from_array {
+    ($aligned:ty, $unaligned:ty, $single:path) => {
+        #[doc = concat!("Convert an array of `", stringify!($aligned), "` to an array of `", stringify!($unaligned), "`.")]
+        pub const fn from_array<const N: usize>(arr: [$aligned; N]) -> [Self; N] {
+            if N == 0 {
+                return unsafe { *(core::ptr::NonNull::dangling().as_ptr() as *const [Self; N]) };
```

Comment:
Nit: You should be able to remove the `unsafe` here; `[]` is a valid `[T; 0]`

Reply:
Unfortunately, the compiler cannot infer that `[]` is of size `N`: https://github.com/unicode-org/icu4x/pull/3445#discussion_r1200841517

Reply:
How about if we invoke `Self::const_default()` (and require that anyone using this macro implements that function)?

I think that's preferable to unsafe code. We should avoid `unsafe` code whenever possible because it triggers additional pain downstream because many organizations have policies where any new `unsafe` code requires manual reviews.

Reply:
Let's go with the `$zero` parameter you had before.

Reply:
Did you have a look at the [original way](https://github.com/unicode-org/icu4x/pull/3445/commits/a7bc4b05dd4c250493252475459a551bb911cd07#diff-c8d7ac875a0d441606621699ac3cb80fbd916d5054d81881076783f1f864ea71R7)? It would save a bit of space (~3 lines vs ~10 chars), and make the requirement _explicit_ (users cannot even think of calling the macro without passing that `$zero` argument).

FWIW the `$zero` should probably be renamed to `$default`, if we end up going that way. 

I have no strong preference, what do you think?

Reply:
Yeah I like the "original way" code. Naming the parameter `$default` seems fine.

Reply:
Was too slow for @robertbastian's comment - ignore my previous comment 👍